### PR TITLE
Np 3491 more org haspart data

### PR DESCRIPTION
**Pull request by svenngjeto**

### CODE REVIEW

File: .github/workflows/karate.yml
Comment (torarnet) [inline .github/workflows/karate.yml:33]: Nice

File: .github/workflows/karate.yml
Comment (torarnet) [inline .github/workflows/karate.yml:39]: Uncomment before merge

File: .github/workflows/karate.yml
Comment (torarnet) [inline .github/workflows/karate.yml:31]: waiting*

File: cristin-commons/src/main/java/no/unit/nva/cristin/model/Constants.java
Comment (brinxmat) [inline cristin-commons/src/main/java/no/unit/nva/cristin/model/Constants.java:48]: URI -> resource

File: cristin-organization/src/test/java/no/unit/nva/cristin/organization/QueryCristinOrganizationHandlerTest.java
Comment (brinxmat) [inline cristin-organization/src/test/java/no/unit/nva/cristin/organization/QueryCristinOrganizationHandlerTest.java:151]: getAttempt? or getTry

File: cristin-organization/src/test/java/no/unit/nva/cristin/organization/organization.feature
Comment (brinxmat) [inline cristin-organization/src/test/java/no/unit/nva/cristin/organization/organization.feature:50]: URI -> resource

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java
Comment (torarnet) [inline cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java:238]: Reuse isSuccessful?

File: cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java
Comment (torarnet) [inline cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java:244]: Error messages is used internally right? I can see you throw a client ApiGatewayException in handleError so I assume this is internal

File: cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java
Comment (torarnet) [inline cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java:88]: Is this the only non successful case?

File: cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java
Comment (torarnet) [inline cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java:88]: Is this error message the one that gets returned to client?

Comment (svenngjeto) [reply cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java:88]: It is beeing rewrappet in the handler, but the message is kept

File: cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java
Comment (svenngjeto) [reply cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java:86]: From cristin we have 404 (Not found, which may be considered a success) and 400 which is error in query statement - and  it might also be any other error from the net

File: cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java
Comment (torarnet) [inline cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java:83]: Is json exception returned to client? Should we return something else?

Comment (svenngjeto) [reply cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java:83]: We go for json-ld as the default

Comment (torarnet) [reply cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java:83]: What I mean is: I see the statuscode is Bad Gateway, but what about the message back to client? Should we return the content of a JsonProcessingException back to the client, or should we hide the internal error and return a more generic one to the client?

File: cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java
Comment (torarnet) [reply cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java:88]: So the client will receive a status code and message from cristin (from the Response object) and not from nva if this fails?

File: cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java
Comment (torarnet) [inline cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java:97]: Can this go into an infinite loop?

File: cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java
Comment (torarnet) [inline cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java:105]: Very similar to `getParentOrganization` . Might not be a problem

File: cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java
Comment (svenngjeto) [reply cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java:83]: I read something else in the first place...  We should hide/camouflage the json exception as something more infomative

**Review by torarnet — COMMENTED** [review]
See other comments, will wait for response before approval.

File: cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java
Comment (svenngjeto) [reply cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java:97]: Only if there is a loop in the data in cristin

File: cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java
Comment (svenngjeto) [reply cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java:105]: It is a tiny difference in getting the siblings children - they are not so interesting when going upwards in the tree

File: cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java
Comment (torarnet) [reply cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java:97]: Good thing we have a timeout in lambda

Comment (svenngjeto) [reply cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java:100]: I runs out of memory, have tried that....

**Review by torarnet — APPROVED** [review]

**Review by brinxmat — APPROVED** [review]

**Pull request merged**
